server/start.sh

```diff
@@ -25,4 +25,4 @@ if [ "$REDIS_PASSWORD_FILE" ]; then
 	unset REDIS_PASSWORD_FILE
 fi
 
-exec node dist/main $1
```

Comment:
this should probably be in the `Dockerfile` instead. Something like

```Dockerfile
ENTRYPOINT ["tini", "--", "sh", "-c"]
```

Reply:
Apart from the machine-learning container, the default entrypoint is `/bin/sh`. Changing this might affect cli or test execution via docker.

Reply:
I think it will be fine, just feels weird to have the script use tini as it's only required for containers.

Reply:
Ideally we should switch all these entrypoints to

```
ENTRYPOINT ["/usr/bin/tini", "--"]
```

The only downside would be that we'd have to tinker with the compose file and adjust the commands for server and machine-learning. e.g. `command: [ "./start.sh", "immich" ]`

Reply:
I agree, but I don't see why any changes are required to the compose file.

Reply:
The current commands in the compose file are missing the `./` prefix. You'd end up with this:

```
tini -- start.sh server
```

As `start.sh` is not on the path this will fail.

Reply:
Completely out of scope for this PR but why don't you create separate images for server and microservices? With the correct usage of `ARG` docker will reuse 99% of the image and we could drop the `command` from our compose file.

https://docs.docker.com/engine/reference/builder/#impact-on-build-caching

Reply:
Because it would be the exact same image. Would be wasteful and more complex for no real benefit.